### ISSUE TITLE
Measure potential gas cost optimisation for switching to another algorithm

### ISSUE THREAD

**Issue by hatskier** [redstone-finance/redstone-oracles-monorepo#2]
Kuba has proposed another algorithm, which allows to get rid of 2-dimensional arrays, but complicates the contract code and requires additional assumptions for the signed data packages (data points must be sorted by their symbols) and requested symbols ordering (they also need to be sorted lexicographically).

We need to measure potential gains in terms of gas to decide if it makes sense to switch to this new algorithm.

**Comment by hatskier** [comment]
Implementation for the memory cost benchmarking is here: https://github.com/redstone-finance/redstone-oracles-monorepo/tree/2-measure-potential-gas-cost-optimisation-for-switching-to-another-algorithm

Benchmarking report is here: https://github.com/redstone-finance/redstone-oracles-monorepo/blob/2-measure-potential-gas-cost-optimisation-for-switching-to-another-algorithm/packages/evm-connector/benchmarks/2022-07-17-with-memory-costs.txt

Summing up, memory allocation in case of the first algorithm consumes only <10% of all related costs (related to the RedStone data extraction and validation).
